services/galley/test/integration/API.hs

```diff
@@ -1239,6 +1253,145 @@ getConvIdsFailMaxSize = do
   getConvIds usr Nothing (Just 1001)
     !!! const 400 === statusCode
 
+listConvIdsOk :: TestM ()
+listConvIdsOk = do
+  [alice, bob] <- randomUsers 2
+  connectUsers alice (singleton bob)
+  void $ postO2OConv alice bob (Just "gossip")
+  let paginationOpts = GetPaginatedConversationIds Nothing (toRange (Proxy @5))
+  listConvIds alice paginationOpts !!! do
+    const 200 === statusCode
+    const (Right 2) === fmap length . decodeQualifiedConvIdList
+  listConvIds bob paginationOpts !!! do
+    const 200 === statusCode
+    const (Right 2) === fmap length . decodeQualifiedConvIdList
+
+paginateConvListIds :: TestM ()
+paginateConvListIds = do
+  [alice, bob, eve] <- randomUsers 3
+  connectUsers alice (list1 bob [eve])
+  localDomain <- viewFederationDomain
+  let qAlice = Qualified alice localDomain
+  now <- liftIO getCurrentTime
+  fedGalleyClient <- view tsFedGalleyClient
+
+  replicateM_ 197 $
+    postConv alice [bob, eve] (Just "gossip") [] Nothing Nothing
+      !!! const 201 === statusCode
+
+  remoteChad <- randomId
+  let chadDomain = Domain "chad.example.com"
+      qChad = Qualified remoteChad chadDomain
+  replicateM_ 25 $ do
+    conv <- randomId
+    let cmu =
+          FederatedGalley.ConversationMemberUpdate
+            { FederatedGalley.cmuTime = now,
+              FederatedGalley.cmuOrigUserId = qChad,
+              FederatedGalley.cmuConvId = Qualified conv chadDomain,
+              FederatedGalley.cmuAlreadyPresentUsers = [],
+              FederatedGalley.cmuUsersAdd = [(qAlice, roleNameWireMember)],
+              FederatedGalley.cmuUsersRemove = []
+            }
+    FederatedGalley.updateConversationMemberships fedGalleyClient cmu
+
+  remoteDee <- randomId
+  let deeDomain = Domain "dee.example.com"
+      qDee = Qualified remoteDee deeDomain
+  replicateM_ 31 $ do
+    conv <- randomId
+    let cmu =
+          FederatedGalley.ConversationMemberUpdate
+            { FederatedGalley.cmuTime = now,
+              FederatedGalley.cmuOrigUserId = qDee,
+              FederatedGalley.cmuConvId = Qualified conv deeDomain,
+              FederatedGalley.cmuAlreadyPresentUsers = [],
+              FederatedGalley.cmuUsersAdd = [(qAlice, roleNameWireMember)],
+              FederatedGalley.cmuUsersRemove = []
+            }
+    FederatedGalley.updateConversationMemberships fedGalleyClient cmu
+
+  -- 1 self conv + 2 convs with bob and eve + 197 local convs + 25 convs on
+  -- chad.example.com + 31 on dee.example = 256 convs. Getting them 16 at a time
+  -- should get all them in 16 times.
+  foldM_ (getChunkedConvs 16 0 alice) Nothing [16, 15 .. 0 :: Int]
+
+-- This test exists
```

Comment:
Are you saying it is a duplicate and of which one?

Reply:
Maybe I started writing the comment and something distracted me :sweat_smile: